src/niscope/system_tests/test_system_niscope.py

```diff
@@ -352,15 +220,6 @@ def test_reset_with_defaults(self, multi_instrument_session):
         multi_instrument_session.reset_device()
```

Comment:
We were supposed to have a version of this test in TestGrpc that checks for the NotImplementedError but it seems to have been deleted somewhere along the way. This test should actually be calling reset_with_defaults, not reset_device. That's why it's passing for both cases, right now. Please split the test into 2 versions and make sure it calls the right method.

Reply:
Turns out it should also be using a single_instrument_session. Multi-instrument sessions do not support niScope_ResetWithDefaults.

Reply:
Fixed this and added tests for grpc and non grpc cases